### PR TITLE
#29 Show current Git version in top-level navigation

### PR DESCRIPTION
**Pull request by WGierke**
For debugging purposes it would be really nice to see in the top right corner which version is currently used. The version is the short SHA-1 commit hash.

## Description
Since I didn't want to add the whole `.git` directory to the Docker container, I just added `.git/logs/HEAD` as a volume which contains historical information about branch changes etc. 

## Reference to official issue
This should solve #29 

## Motivation and Context
The big advantage is that by just adding `HEAD` we don't have the full Git history in our container. Furthermore, every time we switch branches or commit, the `HEAD` file is synchronized with the container (as well as the Docker volumes). In the container, a Django view parses the short current commit hash out of that file and displays it. This way, when switching branches not only the code in the container is synchronized with the code at the host but also the commit hash is updated accordingly. The container can then also be deployed to production.

## How Has This Been Tested?
1. `docker-compose -f local.yml up`  yields
screenshot from 2017-08-24 23-11-18

(with the short version of the commit of this branch: a9f0e6a)
2. add an empty commit: `git commit -m "l" --allow-empty`
3. I refreshed the page and ended up with
screenshot from 2017-08-24 23-12-13
which was the short version of the commit hash of the empty commit (1bd5c33)

## CLA
- [X] I have signed the CLA; if other committers are in the commit history, they have signed the CLA as well

### CODE REVIEW

**Comment by WGierke** [comment]
Any feedback is highly appreciated, especially since I'm a complete novice concerning Docker :)

File: interface/backend/static/views.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from django.shortcuts import render
+
+
+def home(request):
+    current_hash = open('/HEAD', 'r').readlines()[-1].split(' ')[1]
```
Comment (lamby) [inline interface/backend/static/views.py]: So, there is "prior art" within Django for this kind of lookup - I think it's called ``get_version`` or similar. Basically, it needs to a) cache and b) be in some utility method.

(Also, I prefer ``with`` statements; we're relying on garbage collection here to close the file descriptor)

Comment (lamby) [reply interface/backend/static/views.py]: Another approach would be to do this once per instance in ``settings.py`` and assign to VERSION = "whatever".

Comment (lamby) [reply interface/backend/static/views.py]: It also requires some kind of fallback- lots of things clean ".git" etc. etc so falling back to "(unknown)" on any error seems sensible.

Comment (WGierke) [reply interface/backend/static/views.py]: Thanks for your feedback @lamby ! I think I changed everything accordingly. Though, when creating a new commit or switching branches, the version number is not updated. One is now required to restart the docker container but I think that shouldn't be such a big problem?

**Comment by lamby** [comment]
Thanks @WGierke... merged!

**Pull request merged**